docs/discussion/Cosign Upgrade 2024.md

```diff
@@ -0,0 +1,466 @@
+# Cosign Upgrade 2024
+Author: Akash Singhal (@akashsinghal)
+
+Tracked issues in scope:
+- [Support Cosign verification with multiple keys](https://github.com/deislabs/ratify/issues/1191)
+- [Support for Cosign verification with keys managed in KMS](https://github.com/deislabs/ratify/issues/1190)
+- [Support Cosign verification with RSA key](https://github.com/deislabs/ratify/issues/1189)
+
+Ratify currently supports keyless cosign verification which includes an optional custom Rekor server specification. Transparency log verification only occurs for keyless scenarios. Keyed verification is limited to a single public key specified as a value provided in the helm chart. The chart creates a `Secret` for the cosign key and mounts it at a well-known path in the Ratify container. Users must manually update the `Secret` to update the key. There is no support for multiple keys. There is no support for keys stored KMS. There is only support for ECDSA keys, and not RSA or ED25519. There is no support for certificates.
+
+## Support key configuration as K8s resource
+
+Currently, cosign verifier looks for a single key that already exists at a specified path in the Ratify container. Ratify helm chart contains a `Secret` which is mounted at the specified mount point. Ratify's key management experience should be decoupled from secrets and mount paths. Instead, it should be a first class key management experience similar to how certificates are managed via `CertificateStore`.
+
+### Option 1: Extend existing Certificate Store
```

Comment:
I vote for option1 but keep using the CRD name `CertificateStore`. Cert and Key are pairs so I think it's acceptable to add key-related fileds to the CRD.

Reply:
If we consider deprecating the CertificateStore and use the new unified CRD in Ratify v2, that's acceptable even there is a breaking change.